batchflow/tests/regression_metrics_test.py

```diff
@@ -8,33 +8,81 @@
-    for metric_name, exp in exp_res.items():
-        res = metrics.evaluate(metrics=metric_name, agg=output_agg)
-        assert np.allclose(res, exp, atol=0.01, rtol=0), 'failed on metric {}'.format(metric_name)
+PARAMS_DEFINED = [(None, 'mae', [1.5, 0.25]),
+                  (None, 'mse', [3, 0.25]),
+                  (None, 'median_absolute_error', [1, 0.5]),
+                  (None, 'max_error', [2, 1]),
+                  (None, 'rmse', [1.73, 0.5]),
+                  (None, 'r2', [-15, -0.33]),
+                  (None, 'explained_variance_ratio', [-3, 0]),
+                  (None, 'acc', [1, 1]),
+                  ('mean', 'mae', 0.875),
+                  ('mean', 'mse', 1.625),
+                  ('mean', 'median_absolute_error', 0.75),
+                  ('mean', 'max_error', 1.5),
+                  ('mean', 'rmse', 1.12),
+                  ('mean', 'r2', -7.67),
+                  ('mean', 'explained_variance_ratio', -1.5),
+                  ('mean', 'acc', 1)]
+
+@pytest.mark.parametrize('targets, predictions, weights', [([[1, 1], [2, 2]], [[1, 0], [4, 2]], [1, 3])])
+@pytest.mark.parametrize('output_agg, metric_name, exp_res', PARAMS_DEFINED)
+def test_defined_values(targets, predictions, weights, output_agg, metric_name, exp_res):
+    """ Test that expected values match the real ones for some non border values """
+    metrics = RegressionMetrics(targets, predictions, weights=weights, multi=True)
+    res = metrics.evaluate(metrics=metric_name, agg=output_agg)
+    assert np.allclose(res, exp_res, atol=0.01, rtol=0), 'failed on metric {}'.format(metric_name)
+
+
+PARAMS = [('eq', 'mae', 0), ('eq', 'mse', 0), ('eq', 'median_absolute_error', 0), ('eq', 'max_error', 0),
+          ('eq', 'rmse', 0), ('eq', 'r2', 1), ('eq', 'explained_variance_ratio', 1), ('eq', 'acc', 1),
+          ('both', 'mae', 0), ('both', 'r2', np.nan), ('both', 'median_absolute_error', 0), ('both', 'max_error', 0),
+          ('both', 'rmse', 0), ('both', 'mse', 0), ('both', 'explained_variance_ratio', np.nan), ('both', 'acc', 1),
+          ('single', 'r2', -np.inf), ('single', 'explained_variance_ratio', -np.inf)]
+
+@pytest.mark.parametrize('multi', [False, True])
+@pytest.mark.parametrize('mode, metric_name, exp_res', PARAMS)
+def test_both_zero(multi, mode, metric_name, exp_res):
+    """ Test expected values for the cases:
+        1. both targets and predictions are the same non-zeros
+        2. targets and predictions are both zeros
+        3. targets are zeros and predictions are non-zeros
+    """
+    size = (10, 2) if multi else 10
+    exp_res = [exp_res, exp_res] if multi else exp_res
+
+    if mode == 'eq':
+        targets = np.random.randint(10, size=size)
+        predictions = targets
+    elif mode == 'both':
+        targets = np.random.randint(1, size=size)
+        predictions = targets
+    else:
+        targets = np.random.randint(1, size=size)
+        predictions = np.random.randint(10, size=size)
```

Comment:
Having random values in tests is not a good idea since these tests will fail sometimes (even though quite rarely) just because of a mere luck (or rather unluckiness)